### PR TITLE
cloud: adjust toc to align the limits of dedicated and serverless

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

- adjust toc to align the limits of dedicated and serverless
- modify the title of `tidb-cloud/limitations-and-quotas.md`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by zhangyangyu — APPROVED** [review]

File: TOC-tidb-cloud.md [hunk 1]
```diff
@@ -237,7 +237,8 @@
     - [Storage](/tidb-storage.md)
     - [Computing](/tidb-computing.md)
     - [Scheduling](/tidb-scheduling.md)
-  - [TiDB Cloud Cluster Limits and Quotas](/tidb-cloud/limitations-and-quotas.md)
+  - [Dedicated Tier Limits and Quotas](/tidb-cloud/limitations-and-quotas.md)
```
Comment (qiancai) [inline TOC-tidb-cloud.md:240]:
```suggestion
  - [Dedicated Tier Limitations and Quotas](/tidb-cloud/limitations-and-quotas.md)
```

File: tidb-cloud/limitations-and-quotas.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
 ---
-title: Limitations and Quotas in TiDB Cloud
+title: Limitations and Quotas for Dedicated Tier
 summary: Learn the limitations and quotas in TiDB Cloud.
 ---
 
-# Limitations and Quotas in TiDB Cloud
+# Limitations and Quotas for Dedicated Tier
```
Comment (qiancai) [inline tidb-cloud/limitations-and-quotas.md:6]: 要不要也写成 Dedicated Tier Limitations and Quotas，和 Serverless Tier Limitations 的形式对齐

File: tidb-cloud/limitations-and-quotas.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
 ---
-title: Limitations and Quotas in TiDB Cloud
+title: Limitations and Quotas for Dedicated Tier
```
Comment (ran-huang) [inline tidb-cloud/limitations-and-quotas.md:2]:
```suggestion
title: Dedicated Tier Limitations and Quotas
```

File: tidb-cloud/limitations-and-quotas.md (hunk 2 above)
Comment (ran-huang) [inline tidb-cloud/limitations-and-quotas.md:6]:
```suggestion
# Dedicated Tier Limitations and Quotas
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**